### PR TITLE
feature: supports deploying filestash to subfolders of domain names. …

### PR DESCRIPTION
**Pull request by eaglexmw-gmail**
feature: supports deploying filestash to subfolders of domain names. 
like this: www.example.com/filestash.

add new env var FILESTASH_BASEURL for support nginx reverse config:
    proxy_pass http://xxx:8334/;
or
   proxy_pass http://xxx:8334;

### CODE REVIEW

**Comment by mickael-kerjean** [comment]
I would have hope you reach out before making a PR. That [feature](https://github.com/mickael-kerjean/filestash/blob/master/public/index.frontoffice.html#L4) is already supported in the canary release. To get it to work, you need to use 2 variable: "CANARY=true" to use the new frontend and "FILESTASH_BASE=/whatever/" to change the application base url to whatever you want. The client folder is deprecated, a lot of work has been into trying to get rid of it in favor of the new one which will get release soon

**Comment by eaglexmw-gmail** [comment]
My opinion, my implementation is better. 
First of all, in my implementation, a file can support both Filestash deployment in the root directory and subdirectories, without the need to provide additional front-end files; Perfectly aligned with golang's web service philosophy, a file is a service.
Second, in my implementation, due to the convergence of the front and back ends, only a small number of code modifications need to be considered for deployment differences, and the rest of the code is generic and the modifications are compatible with various situations, so it can also maximize efficiency.
Finally, regarding the changes to the index.frontend.html file, I'm not sure if the code is still in use, possibly, it needs to be removed?

**Comment by eaglexmw-gmail** [comment]
@mickael-kerjean

**Comment by eaglexmw-gmail** [comment]
@mickael-kerjean

> > My opinion, my implementation is better.
> 
>     1. this code break the existing implementation which is a nono without a proper transition plan
> 
>     2. this code makes change on the client folder which will be deleted in the next couple months in favor of the new frontend that has been baking for more than a year.
> 
>     3. this code rely on the old legacy webpack config when the new release doesn't use it
> 
>     4. seeing `window.globalConfig.BaseUrl` everywhere is to me a code smell considering the [base html](https://www.w3schools.com/TAGs/tag_base.asp) tag is a standard. I prefer using standard than making assumption around global window objects. The current implementation of that feature heavily relies on that base tag
Yes, use base html is the better idea.
> 
> 
> > First of all, in my implementation, a file can support both Filestash deployment in the root directory and subdirectories, without the need to provide additional front-end files. Second, in my implementation, due to the convergence of the front and back ends, only a small number of code modifications need to be considered for deployment differences, and the rest of the code is generic and the modifications are compatible with various situations, so it can also maximize efficiency.
> 
> I'm not a native english speaker and I can't understand what you are talking about. Can you rephrase it and give concrete examples of what you want to improve?
I just want Filestash to be deployed both at the root or in subdirectories, not need to rely on other external resources.
I think it's a great idea.

> 
> > Finally, regarding the changes to the index.frontend.html file, I'm not sure if the code is still in use, possibly, it needs to be removed?
> 
> Yes, it's heavily used, running in production and your code is breaking it all for those who rely on it
Unfortunately, I misunderstood the direction of the product, because in the current recent code, the code under the client directory is still compiled.

OK, I will close this pull request.

**Comment by eaglexmw-gmail** [comment]
wait new frontend.
